src/main/java/de/tum/in/www1/artemis/service/connectors/GitService.java

```diff
@@ -469,7 +469,7 @@ public void combineAllStudentCommits(Repository repository, ProgrammingExercise
 
             studentGit.reset().setMode(ResetCommand.ResetType.SOFT).setRef(latestHash.getName()).call();
             studentGit.add().addFilepattern(".").call();
-            var student = ((StudentParticipation) repository.getParticipation()).getStudent();
+            var student = ((StudentParticipation) repository.getParticipation()).getStudents().iterator().next();
```

Comment:
Great catch. In theory it would indeed be possible that someone removed all students from a team that already has programming exercise participations before this method is called. Then this would have thrown a `NoSuchElementException`. I fixed it so that it uses the existing fallback correctly for empty teams.